firebase-web/src/test/java/io/spine/web/firebase/rest/HttpClientMockFactory.java

```diff
@@ -0,0 +1,131 @@
+
+/**
+ * Creates mocked {@link HttpClient}s.
+ */
+final class HttpClientMockFactory {
+
+    /**
+     * Prevents instantiation of this utility class.
+     */
+    private HttpClientMockFactory() {
+    }
+
+    /**
+     * Creates a new no-operation {@code HttpClient}.
+     */
+    static HttpClient noOpClient() {
+        MockHttpTransport transport = new MockHttpTransport.Builder().build();
+        return HttpClient.using(transport.createRequestFactory());
+    }
+
+    /**
+     * Creates an {@code HttpClient} mock which returns the specified {@code content}
+     * on every request.
+     */
+    static HttpClient mockHttpClient(String content) {
+        return mockHttpClient(content, HttpClientMockFactory::noOpObserver);
+    }
+
+    /**
+     * Creates an {@code HttpClient} mock which returns the specified {@code content}
+     * on every request and uses supplied {@code observer} while building requests.
+     */
+    static HttpClient mockHttpClient(String content, RequestObserver observer) {
+        return HttpClient.using(mockRequestFactory(content, observer));
+    }
+
+    private static HttpRequestFactory
+    mockRequestFactory(String content, RequestObserver observer) {
+        HttpTransport transportMock = mockHttpTransport(content, observer);
+        HttpRequestFactory requestFactoryMock = transportMock.createRequestFactory();
+        return requestFactoryMock;
+    }
+
+    private static HttpTransport
+    mockHttpTransport(String content, RequestObserver observer) {
+        final MockLowLevelHttpResponse response = new MockLowLevelHttpResponse();
+        response.setContent(content);
+        return new MockHttpTransport() {
+            @Override
+            public LowLevelHttpRequest
+            buildRequest(String method, String url) {
+                MockLowLevelHttpRequest request = new MockLowLevelHttpRequest(url);
+                request.setResponse(response);
+                observer.onRequest(method, url);
+                return request;
+            }
+        };
+    }
+
+    /**
+     * Creates an {@code HttpClient} mock which throws {@link java.io.IOException} on every
+     * request.
+     */
+    static HttpClient throwingClient() {
+        return HttpClient.using(throwingRequestFactory());
+    }
+
+    private static HttpRequestFactory throwingRequestFactory() {
+        HttpRequestFactory result = throwingHttpTransport().createRequestFactory();
+        return result;
+    }
+
+    private static MockHttpTransport throwingHttpTransport() {
+        return new MockHttpTransport() {
+            @Override
+            public LowLevelHttpRequest
+            buildRequest(String method, String url) throws IOException {
+                throw new IOException("Intended exception");
+            }
+        };
+    }
+
+    private static void noOpObserver(String method, String url) {
+    }
+
+    /**
+     * Observes built HTTP requests.
+     */
+    interface RequestObserver {
```

Comment:
This name confuses me as well as the API which uses it. 

If I'd wake somebody up at 3 AM and ask what a `RequestObserver` does, they would hardly answer correctly.